src/conventions/h-group/clue-interpretation/interpret-clue.js

```diff
@@ -297,6 +302,32 @@ export function interpret_clue(game, action) {
 			}
 		}
 
+		// If there's a valid connection that doesn't require a bluff, a bluff is not a valid interpretation.
+		const no_bluff_connections = all_connections.filter(connection => connection.connections.length > 0 && !connection.connections[0].bluff);
+		if (no_bluff_connections.length > 0) {
+			// Convert possible bluff connections to non-bluff connections.
+			all_connections = all_connections.map(conn => {
+				if (!conn.connections[0]?.bluff)
+					return conn;
+				// If not a hidden connection, and we know the bluff card doesn't match, the real card wasn't found.
+				if (!conn.connections[0].hidden && !conn.connections[0].identities.find(i => conn.connections[0].card.matches(i, {assume: true})))
+					return null;
+				conn.connections[0].bluff = false;
+				return conn;
+			}).filter(conn => !!conn);
+		} else {
+			// Filter plays after hidden bluff connection.
+			all_connections = all_connections.map(conn => {
+				if (!conn.connections[0]?.bluff || !conn.connections[0].hidden)
+					return conn;
+				// Remove everything after the bluff play to the non-hidden play as they won't
+				// play after the bluff play.
+				const next_visible_connection = conn.connections.findIndex(c => !c.bluff && !c.hidden);
+				conn.connections.splice(1, next_visible_connection);
```

Comment:
Should this be `next_visible_connection - 1`? If the next real connection is at index 2, then I think we only want to remove 1 element, but this will remove 2.

Reply:
> Should this be `next_visible_connection - 1`? If the next real connection is at index 2, then I think we only want to remove 1 element, but this will remove 2.

No, I think we do want to remove the "real" connection if the first card is a legitimate bluff, as it can't be known that this real connection exists. E.g. if you are bluffed to play an r2 and it doesn't match the real one won't be known.

> Will there be later connections we need to preserve in the first place (maybe when bluffing through already-clued cards)?

Yes, my thinking was that a bluff can reveal to other players that they have cards that resulted in the bluff being valid. E.g. learning that your red card is in fact a red 3 to make a bluff for the r2 work by clueing an r4.

Reply:
I think the only real case where this could happen is when it comes from the same hand. E.g.

Alice clues red to Cathy,
Bob's hand is b1, r1

Bob is going to stop playing after the b1, so we want to erase the r1 connection.